packages/react-app/pages/admin/newBatches.jsx

```diff
@@ -0,0 +1,373 @@
+  const isLoggedIn = userRole !== null && userRole !== USER_ROLES.anonymous;
+  const { baseColor } = useCustomColorModes();
+
+  const ensFiltering = (rows, id, filterValue) => {
+    if (filterValue.length < 3) {
+      return rows;
+    }
+
+    return rows.filter(row => {
+      const rowValue = row.values[id];
+      return rowValue !== undefined ? isValueOnEnsOrSocials(rowValue, filterValue) : true;
+    });
+  };
+
+  const batchFiltering = (rows, id, filterValue) => {
+    if (filterValue === "allBuilders" || filterValue === undefined) {
+      return rows;
+    }
+
+    if (filterValue === "allBatches") {
+      return rows.filter(row => row.values[id]?.batch?.number);
+    }
+
+    return rows.filter(row => {
+      const rowValue = row.values[id];
+      return rowValue !== undefined ? isInBatch(rowValue, filterValue) : true;
+    });
+  };
+
+  useEffect(() => {
+    async function fetchBuilders() {
+      setIsLoadingBuilders(true);
+      const fetchedBuilders = await axios.get(serverUrl + serverPath);
+
+      const processedBuilders = fetchedBuilders.data
+        .filter(builder => builder.ens !== "austingriffith.eth")
+        .map(builder => ({
+          builder,
+          status: builder.status,
+          stream: builder,
+          builds: builder.builds?.length || 0,
+          socials: builder,
+          userCreated: builderCreated(builder),
+        }));
+
+      setBuilders(processedBuilders);
+      setIsLoadingBuilders(false);
+    }
+
+    fetchBuilders();
+  }, [serverUrl]);
+
+  const BatchFilterComponent = useMemo(() => {
+    const Component = ({ column }) => (
+      <BatchColumnFilter filterValue={column.filterValue} setFilter={column.setFilter} builders={builders} />
+    );
+    Component.displayName = "BatchFilterComponent";
+    return Component;
+  }, [builders]);
+
+  const columns = useMemo(
+    () => [
+      getColumn(
+        "Builder",
+        "builder",
+        ({ value }) => <BuilderAddressCell builder={value} mainnetProvider={mainnetProvider} />,
+        { disableSortBy: true, Filter: EnsColumnFilter, filter: ensFiltering },
+      ),
+      getColumn("Status", "status", ({ value }) => <BuilderStatusCell status={value} />, { disableSortBy: true }),
+      getColumn("Builds", "builds", ({ value }) => <BuilderBuildsCell buildCount={value} />, { sortDescFirst: true }),
+      getColumn(
+        "Stream",
+        "stream",
+        ({ value }) => (value?.graduated?.status ? "" : <StreamTableCell builder={value} />),
+        { disableFilters: true, Filter: BatchFilterComponent, filter: batchFiltering },
+      ),
+      getColumn("Socials", "socials", ({ value }) => <BuilderSocialLinksCell builder={value} isAdmin={isAdmin} />, {
+        disableSortBy: true,
+      }),
+      getColumn("User Created", "userCreated", ({ value }) => <DateWithTooltip timestamp={value} />, {
+        sortAscFirst: true,
+      }),
+    ],
+    [BatchFilterComponent, mainnetProvider, isAdmin],
+  );
```

Comment:
Any reason we are using this format here? (instead of just an allColimns object as in /builders)?

![image](https://github.com/user-attachments/assets/e1a44780-038e-47a4-ba05-d841f65ec9e9)

Reply:
I thought it would be maybe nicer to have it in a function to have less code, but let me change it back.